supertokens_python/recipe/emailpassword/recipe.py

```diff
@@ -49,26 +59,45 @@
 class EmailPasswordRecipe(RecipeModule):
     recipe_id = 'emailpassword'
     __instance = None
+    email_delivery_ingredient: EmailDeliveryIngredient[TypeEmailPasswordEmailDeliveryInput]
 
     def __init__(self, recipe_id: str, app_info: AppInfo,
                  sign_up_feature: Union[InputSignUpFeature, None] = None,
                  reset_password_using_token_feature: Union[InputResetPasswordUsingTokenFeature, None] = None,
                  email_verification_feature: Union[InputEmailVerificationConfig, None] = None,
                  override: Union[InputOverrideConfig, None] = None,
-                 email_verification_recipe: Union[EmailVerificationRecipe, None] = None):
+                 email_verification_recipe: Union[EmailVerificationRecipe, None] = None,
+                 email_delivery: Union[EmailDeliveryConfig[TypeEmailPasswordEmailDeliveryInput], None] = None,
+                 ingredients: Union[EmailPasswordIngredients, None] = None
+                 ):
         super().__init__(recipe_id, app_info)
         self.config = validate_and_normalise_user_input(self, app_info, sign_up_feature,
                                                         reset_password_using_token_feature,
-                                                        email_verification_feature, override)
+                                                        email_verification_feature, override, email_delivery)
+        recipe_implementation = RecipeImplementation(Querier.get_instance(recipe_id))
+        self.recipe_implementation = recipe_implementation if self.config.override.functions is None else \
+            self.config.override.functions(recipe_implementation)
+
+        self.recipe_implementation = cast(RecipeImplementation, self.recipe_implementation)
```

Comment:
`get_email_delivery_config` looks for `emailpassword.recipe_implementation.RecipeImplementation`

But the type checkers (like mypy) feel that `recipe_implementation` can be `RecipeInterface`, which for them is different from `RecipeImplementation`

After thinking a little bit, I've now changed it to:
```python
if not isinstance(self.recipe_implementation, RecipeImplementation):
            raise Exception("Should never come here")
```
instead of `self.recipe_implementation = cast(RecipeImplementation, self.recipe_implementation)`

Reply:
Nevermind. The type of `get_email_delivery_config` is now `Callable[[RecipeInterface], EmailDeliveryConfigWithService[TypeEmailPasswordEmailDeliveryInput]]`. We can/should use `RecipeInterface` not `RecipeImplementation`. 
No need of this:
```
if not isinstance(self.recipe_implementation, RecipeImplementation):
            raise Exception("Should never come here")
```